### ISSUE TITLE
Sub Form Field styles problems

### ISSUE THREAD

**Issue by hfittipaldi** [joomla/joomla-cms#17589]
### Steps to reproduce the issue
Any subform with layout="joomla.form.field.subform.repeatable-table"

### Expected result
The form style similar to others. Something like this:
![screenshot-joomla-subform-repeatable-field-type](https://user-images.githubusercontent.com/2341857/29430170-c35a58e0-8369-11e7-80a1-e3e1c6be5b3d.jpg)

### Actual result
A little bit unformatted:
![screenshot-joomla-subform-field-type](https://user-images.githubusercontent.com/2341857/29430022-2a3ea59e-8369-11e7-944f-12a2442e72fc.jpg)

### System information (as much as possible)
PHP executando em – Windows NT 10.0 build 15063 (Windows 10) AMD64
Versão do Banco de Dados – 5.7.14
Versão do PHP – 5.6.30
Servidor Web – Apache/2.4.23 (Win64) OpenSSL/1.0.2j PHP/5.6.30
Versão do Joomla – Joomla! 3.7.4 Stable [ Amani ] 25-July-2017 11:11 GMT
Navegador do Usuário – Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.90 Safari/537.36

**Comment by tonypartridge** [comment]
So the table outputs in a table fashion. It looks like you just want repeatable.

If this isn't the case provide your whole JForm xml so we can see how you are creating it

**Comment by Fedik** [comment]
I can confirm, the styling of table layout `joomla.form.field.subform.repeatable-table` looks really bad.
It has become bad after "flat styling" changes, or around this.

**Comment by tonypartridge** [comment]
Working ok here... I styled a table yesterday.

Is it only a problem in the backend?

-- 
Tony Partridge

From: Fedir Zinchuk <notifications@github.com> <notifications@github.com>
Reply: joomla/joomla-cms
<reply@reply.github.com>
<reply@reply.github.com>
Date: 20 August 2017 at 17:08:32
To: joomla/joomla-cms <joomla-cms@noreply.github.com>
<joomla-cms@noreply.github.com>
CC: Tony Partridge <admin@xtech.im> <admin@xtech.im>, Comment
<comment@noreply.github.com> <comment@noreply.github.com>
Subject:  Re: [joomla/joomla-cms] Sub Form Field styles problems (#17589)

I can confirm, the styling of table layout

**Comment by Fedik** [comment]
ok, it better in recent update 😄 
but still some issue, try:
```xml
<field name="sebform1" type="subform" label="subform"
	layout="joomla.form.field.subform.repeatable-table" multiple="true" groupByFieldset="true">
	<form>
		<fieldset name="group1" label="group 1">
			<field type="text" name="text" label="text 1"/>
			<field type="text" name="text2" label="text 2"/>
			<field type="text" name="text3" label="text 3"/>
		</fieldset>
		<fieldset name="group2" label="group 2">
			<field type="text" name="text4" label="text 4"/>
			<field type="textarea" name="textarea" label="textarea"/>
		</fieldset>
	</form>
</field>
```

![screen 2017-08-20 19 42 10 1161x278](https://user-images.githubusercontent.com/1568198/29496653-b774e496-85df-11e7-8521-47244f8091f2.png)

yes, in the backend

**Comment by ghost** [comment]
@Fedik can you create a Pull Request for mentioned Issue?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17589">issues.joomla.org/tracker/joomla-cms/17589</a>.</sub>

**Comment by Fedik** [comment]
I do not have a fix currently,

**Comment by ghost** [comment]
sorry, now i read again you have an Issue, not PR.

**Comment by hfittipaldi** [comment]
@Fedik , besides the style problem, do you also have problems with remove an empty field? I'm getting javascript error (Joomla 3.8.3 and 3.8.4-rc)

**Comment by Fedik** [comment]
@hfittipaldi  I do not have the problem with remove an empty field

**Comment by hfittipaldi** [comment]
Weird. I'm getting "Uncaught TypeError: Cannot read property 'id' of undefined" when I remove an empty record (not field, my bad)

**Comment by Quy** [comment]
It is because of line 7541 `\administrator\templates\isis\css\template.css` zeroing the bottom margin. I am not sure of the side-effects to removing it.
```
.subform-table-layout .control-group,
.subform-table-layout .control-group:last-of-type {
	margin-bottom: 0;
}
```
![subform](https://user-images.githubusercontent.com/368084/39073486-ccd92cac-44a2-11e8-86c8-ee0ef2ea0ac7.jpg)

**Comment by ciar4n** [comment]
This CSS was originally added here... https://github.com/joomla/joomla-cms/pull/13769

Maybe something has changed since as the issue does not appear to be present in the screenshot of that pr. I'll have a look later.

**Comment by ciar4n** [comment]
Nothing has changed. Looks like I hadn't considered multiple field in each cell when creating the PR. I'll fix this.

**Comment by ciar4n** [comment]
PR created.. https://github.com/joomla/joomla-cms/pull/20209

**Comment by ghost** [comment]
closed as having Pull Request #20209
